content/blog/2019-11-mio-v0.7-alpha.md

```diff
@@ -0,0 +1,201 @@
+
+One use case the user space queue support was waking a thread polling (i.e.
+calling `Poll::poll`) from another thread. To support this use case a new
+`Waker` type was introduced to do exactly this job, calling `Waker::wake` will
+wake the `Poll` it is associated with.
+
+[crossbeam]: https://crates.io/crates/crossbeam
+
+
+### Registering & I/O resource usages changes
+
+By far the biggest user facing change of version 0.7 is the way Mio registers
+I/O sources. In version 0.6 resource would be registered using a `Token`,
+`Ready` interests and `PollOpt`ions. For example the following code would register
+`socket` with readable interests and edge triggers.
+
+```rust
+poll.register(&socket, Token(0), Ready::readable(), PollOpt::edge())?;
+```
+
+As mentioned above the `Event` type was changed to be a wrapper around the
+native OS type, in turn this removed the `Ready` type in favour of having
+methods on `Event` to check for the readiness indicators and getting the
+`Token`. The `Ready` type, as used in registering, was changed to `Interests` to
+better reflect its usage. The API of `Interests` was also changed to take
+advanced of the (somewhat) new associated constants, ensuring that is no longer
+possible to register an event source with empty interests.
+
+In version 0.7 Mio registers all source with edge triggers, removing the need
+for `PollOpt`. How to use edge triggers is explained below.
+
+The trait that defined how to register an event source in version 0.6 was called
+`Evented`, this was changed to `Source` and is now referenced to as
+`event::Source`, as the type lives inside the `event` module. The same three
+methods live on the trait but are changed as described above.
+
+Finally the register functions have moved from `Poll` to a new `Registry` type
+which can be `try_clone`d and used to register sources from different threads.
+All together that means registering the same `socket` with readable interests
+now looks like this:
+
+```rust
+poll.registry().register(&socket, Token(0), Interests::READABLE)?;
+```
+
+
+#### Moving to edge triggers
+
+As already mentioned Mio now registers all I/O source with edge triggers which
+means that users of one-shot and level trigger need to change how they respond
+to events.
+
+The problem was that certain usages of level and one-shot triggers showed
+differences in OSes that Mio couldn't, without introducing a lot of overhead,
+smooth over to make Mio a nice cross-platform experience. Instead we decided to
+change all triggers to be edge triggers, which perform the same way on all
+platforms.
+
+Users that previously used one-shot triggers should now `deregister` the I/O
+source if they no longer want to receive events for it. And once more events are
+wanted the resource should be `register`ed again.
+
+Users that used level triggers basically need to put a loop around all I/O
+operations. When using edge triggers the user is responsible for attempting the
```

Comment:
I suggest:

When using edge triggers the user is required to perform I/O after receiving an event.  For example, in response to a read event, the user must read from the I/O source until it would block(an
`io::Error` of kind `io::ErrorKind::WouldBlock`).